src/detect-byte.c

```diff
@@ -32,25 +32,46 @@
  *
  * \param arg The name of the variable being sought
  * \param s The signature to check for the variable
+ * \param strict Match if and only iff the list sought and the list found equal.
  * \param sm_list The caller's matching buffer
  * \param index When found, the value of the slot within the byte vars
  *
  * \retval true A match for the variable was found.
  * \retval false
  */
 bool DetectByteRetrieveSMVar(
-        const char *arg, const Signature *s, int sm_list, DetectByteIndexType *index)
+        const char *arg, const Signature *s, bool strict, int sm_list, DetectByteIndexType *index)
 {
-    SigMatch *bed_sm = DetectByteExtractRetrieveSMVar(arg, sm_list, s);
+    bool any = sm_list == -1;
+    int found_list;
+    SigMatch *bed_sm = DetectByteExtractRetrieveSMVar(arg, &found_list, s);
     if (bed_sm != NULL) {
+        if (!any && sm_list != found_list) {
+            if (strict) {
+                return false;
+            }
+            SCLogWarning("Using byte variable from a different buffer may produce indeterminate "
+                         "results; variable: \"%s\"",
```

Comment:
@jufajardini should this warning have a link where the rule writer can understand better the risk of FN ?

Reply:
I think that'd be great, indeed.